<div id="about"></div>
# Matthew Shirley, Ph.D.
<div id="contact" class="row">
    <div class="col-sm-8 col-md-8 col-sm-offset-3 col-md-offset-3 well well-sm text-right">
        Department of Oncology<br>  
        Sidney Kimmel Comprehensive Cancer Center<br>  
        The Johns Hopkins School of Medicine<br>  
        CRBII 142<br>  
        1550 Orleans Street<br>  
        Baltimore MD 21287<br>  
        <a href="mailto:matt.shirley@jhmi.edu">matt.shirley@jhmi.edu</a><br>
        <div class="btn-group">
            <button type="button" class="btn btn-primary" onclick="window.location.href = 'mailto:matt.shirley@jhmi.edu'"><span class="glyphicon glyphicon-envelope"></span></button>
            <button type="button" class="btn btn-primary" onclick="window.location.href = 'http://www.google.com/webhp?hl=en&sa=N&tab=lw#q=matt+shirley+mdshw5&hl=en&ei=dCf3UfqpOqHl4AOI1ICgDg&start=0&sa=N&bav=on.2,or.r_cp.r_qf.&fp=2932c87ff975568f&biw=1159&bih=730&bvm=pv.xjs.s.en_US.9jgl75mduIg.O%2Cpv.xjs.s.en_US.9jgl75mduIg.O'"><span class="glyphicon glyphicon-search"></span></button>
            <button type="button" class="btn btn-primary" onclick="window.location.href = 'http://mattshirley.com/about?print=true'"><span class="glyphicon glyphicon-print"></span></button>
        </div>
    </div>
</div>

<div id="education"></div>
### Education
<li class="list-unstyled">
    <ul><strong>2013</strong> - Ph.D. Biochemistry, Cellular, and Molecular Biology, 
  Johns Hopkins School of Medicine, Baltimore MD</ul>
    <ul><strong>2008</strong> - B.S. Biochemistry, University of Missouri, Columbia MO</ul>
</li>

<div id="research" style="page-break-after:always;"></div>
### Research

- **2013-Current** - Postdoctoral Fellow, Department of Oncology,
 Johns Hopkins School of Medicine, Baltimore MD

    *Working in the labs of Drs. Sarah Wheelan and Vasan Yegnasubramanian I
     developed computational tools for the analysis of novel high-dimensional
     sequencing methods. We applied these tools to prostate cancer biomarker
     discovery.*

- **2008-2013** - Ph.D. Candidate, Department of Biochemistry,
 Cellular, and Molecular Biology, Johns Hopkins School of Medicine, Baltimore MD

    *As a graduate student I started research with Drs. Jin Zhang and
     Heng Zhu in the Department of Pharmacology at Johns Hopkins School
     of Medicine. Using the protein microarrays pioneered by the Zhu lab
     we interrogated the kinome for novel kinase-substrate interactions.
     I continued my graduate research in the lab of Dr. Jonathan Pevsner
     at Kennedy Krieger Institute where I characterized the extent of
     genetic variability in lymphoblast cell lines.
     My research culminated in the discovery of a somatic mutation in
     **GNAQ** which is found only in affected tissues from patients
     with the Sturge-Weber syndrome. This work spanned wet-lab and
     computational work involving microarray design and analysis as
     well as DNA resequencing assay design, high-dimensional data
     analysis, and tool development.*

- **2006-2008** - Research Assistant, Department of Biochemistry,
 University of Missouri, Columbia MO

    *In my third year of college at the University of Missouri
     I joined the lab of Dr. Judy Wall in the Department of Biochemistry.
     In the Wall lab I worked on characterizing the molecular genetics of
     the bacterium **Desulfovibrio vulgaris** Hildenborough for its potential
     in heavy metal bioremediation. My work involved characterizing a Tn5
     transposon mediated mutant library, as well as operation and configuration
     of a gas chromatagraph.*

<div id="publications"></div>
### Peer-Reviewed Publications

*See a full of publications at [Google Scholar](http://scholar.google.com/citations?user=b7Jyb4YAAAAJ&hl=en).*

- **Shirley MD**, Tang H, Gallione CJ, Baugher JD, Frelin LP, Cohen B, North PE, Marchuk DA, Comi AM, Pevsner J. *[Sturge–Weber Syndrome and Port-Wine Stains Caused by Somatic Mutation in GNAQ][3]*. New England Journal of Medicine. 2013. PMID: [23656586][4].  

- R.H. Newman, J. Hu, H. Rho, Z. Xie, C. Woodard, J. Neiswinger, C. Cooper, **M. Shirley** et al. *[Construction of human activity-based phosphorylation networks][13]*. Molecular Systems Biology. 2013. PMID: [23549483][14] [Open Access][15].

- Stevens EL, Baugher JD, **Shirley MD**, Frelin LP, Pevsner J. *[Unexpected Relationships and Inbreeding in HapMap Phase III Populations][16]*. PLoS ONE. 2012. PMID: [23185369][17].

- **Shirley MD**, Baugher JD, Stevens EL, Tang Z, Gerry N, Beiswanger CM, Berlin DS, Pevsner J. *[Chromosomal variation in lymphoblastoid cell lines][18]*. Human Mutation. 2012. PMID: [22374857][19]

<div id="software"></div>
### Software

- [pyfaidx: efficient pythonic random access to fasta subsequences](http://dx.doi.org/10.5281/zenodo.8548). ZENODO. 10.5281/zenodo.8548
- [fastqp](https://github.com/mdshw5/fastqp): Simple FASTQ and SAM read quality assessment and plotting using Python.
- [hamstring][9]: Python module for generating Hamming7,4 error-correcting DNA barcodes. Based on work by [Leonid Bystrykh][10].
- [triPOD][11]: web interface for the triPOD analysis developed by [Joe Baugher et al][12].
- [sra toolkit](http://toolshed.g2.bx.psu.edu/view/matt-shirley/ncbi_sra_toolkit): Galaxy tool definitions for [NCBI SRA Toolkit](http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software)
- [ncbi epigenomics](http://toolshed.g2.bx.psu.edu/view/matt-shirley/ncbi_epi_browse): Galaxy interface to [NCBI Epigenomics](http://www.ncbi.nlm.nih.gov/epigenomics)

<div id="grants"></div>
### Grants
- 2013 - AWS in Education Grant, supporting *Introduction to Python* short course in the Center for Computational Genomics

<div id="teaching"></div>
### Teaching

- **2013-Current** - Instructor, Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD
- **2013-Current** - Instructor, Bioinformatics & Computational Biology Department, Foundation for Advanced Education in the Sciences, Bethesda MD

<div class="hidden-print">
    <h5>Titles of talks link to slides.</h5>
</div>

- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][29], April 8, 2014, FAES, Bethesda MD
- Introduction to Python^: [data structures and flow control](uploads/2013/10/Introduction%20to%20Python%20Day%201.ipynb.slides.html), October 28 2013, Center for Computational Genomics, Baltimore MD
- Introduction to Python^: [functions, classes, and I/O](uploads/2013/10/Introduction%20to%20Python%20Day%202.ipynb.slides.html), October 30 2013, Center for Computational Genomics, Baltimore MD  
^ *Supported by AWS in Education Grant award*  
- BIOF503: Make Striking Figures and Analyze Data with R, [Spoon-Fed R](uploads/2013/10/Spoon-fed_R.html), October 24 2013, FAES, Bethesda MD
- BIOF309: Introduction to Python, [Interfacing R and Python using Rpy2][22], May 2 2013, FAES, Bethesda MD
- BIOF309: Introduction to Python, [Hello web: Accessing databases and APIs using Biopython][24], April 25 2013, FAES, Bethesda MD
- BIOF309: Introduction to Python, [Reading, processing, and writing high throughput sequencing data][25], April 18 2013, FAES, Bethesda MD
- BIOF521: Bioinformatics for analysis of data generated by NGS, [Galaxy for Next Generation Sequencing Analysis][26], February 6 2013, FAES, Bethesda MD
- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][29], July 9 2013, FAES, Bethesda MD
- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][28], November 12 2012, FAES, Bethesda MD

<div id="seminars"></div>
### Seminars

- Computational Genomics Young Investigators Symposium, [Deep amplicon sequencing reveals GNAQ 548G>A as the causal somatic mutation in Sturge-Weber syndrome and common port-wine stains](uploads/2013/10/Young-Investigators-Seminar.pdf), October 17 2013, Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD
- Thesis Seminar, [Sturge–Weber syndrome & port-wine stain: Genomic analysis reveals somatic activating mutation in GNAQ as genetic basis for neurocutaneous disorder][21], May 22 2013, Department of Biochemistry, Cellular, and Molecular Biology, Johns Hopkins School of Medicine, Baltimore MD
- Graduate Student Colloquium, [Chromosomal Variation in Lymphoblastoid Cell Lines][27], December 10 2012, Department of Biochemistry, Cellular, and Molecular Biology, Johns Hopkins School of Medicine, Baltimore MD

<div id="posters" style="page-break-after:always;"></div>
###Posters

- Graduate Student Association Poster Session, [Chromosomal Variation in Lymphoblastoid Cell Lines][39], April 25 2012, Johns Hopkins School of Medicine, Baltimore MD
- Undergraduate Research Day at the Capitol, [Rapid Automated Characterization of Transposon Insertion Mutants in *Desulfovibrio vulgaris* Hildenborough by srnPCR][30], April 2008, Jefferson City MO
- Undergraduate Research and Creative Achievements Forum, [Rapid Automated Characterization of Transposon Insertion Mutants in *Desulfovibrio vulgaris* Hildenborough by srnPCR][30], 2007, Winner of Chancellor’s Award for Excellence in Research, University of Missouri, Columbia MO
- Environmental Stress Pathway Project, [Rapid Automated Characterization of Transposon Insertion Mutants in *Desulfovibrio vulgaris* Hildenborough by srnPCR](http://vimss.lbl.gov/publicfiles/ESPP_retreat_2007_Shirley.pdf), 2007, Virtual Institute for Microbial Stress and Survival, Lawrence Berkeley National Laboratories, Berkeley CA

<div id="awards"></div>
### Awards
- 2007 - Chancellor’s Award for Excellence in Research, Undergraduate Research and Creative Achievements Forum, University of Missouri, Columbia MO

 [1]: http://biolchem.bs.jhmi.edu/bcmb/Pages/index.aspx
 [2]: http://biochem.missouri.edu/
 [3]: uploads/2012/03/N-Engl-J-Med-2013-Shirley.pdf
 [4]: http://www.ncbi.nlm.nih.gov/pubmed/23656586
 [5]: http://www.sturge-weber.org/component/content/article/4-general-info/311-route-to-a-cure.html
 [6]: http://www.sturge-weber.org/images/stories/BVMC/explaining_gnaq.pdf
 [7]: http://www.sturge-weber.org/images/stories/BVMC/sturge-weber%20foundation%20nejm%20news%20release%20final%20may-08-2013.pdf
 [8]: http://faes.org
 [9]: http://mdshw5.github.io/hamstring/
 [10]: http://www.plosone.org/article/info:doi/10.1371/journal.pone.0036852
 [11]: http://tripod.mattshirley.com
 [12]: http://www.biomedcentral.com/1471-2164/14/367/
 [13]: uploads/2013/04/Mol-Syst-Biol-2013-Newman.pdf
 [14]: http://www.ncbi.nlm.nih.gov/pubmed/?term=23549483
 [15]: http://www.nature.com/doifinder/10.1038/msb.2013.12
 [16]: uploads/2013/03/journal.pone_.0049575.pdf
 [17]: http://www.ncbi.nlm.nih.gov/pubmed/23185369
 [18]: uploads/2012/03/22062_ftp.pdf
 [19]: http://www.ncbi.nlm.nih.gov/pubmed/22374857
 [21]: uploads/2013/05/SWS_thesis_seminar.html
 [22]: uploads/2013/04/lecture14-notes.html
 [23]: http://www.faes.org
 [24]: uploads/2013/04/lecture13-notes.html
 [25]: uploads/2013/04/lecture12-notes.html
 [26]: uploads/2013/02/Galaxy-for-NGS-Analysis.pdf
 [27]: uploads/2012/10/Chromosomal-Variation-in-Lymphoblastoid-Cell-Lines.pdf
 [28]: uploads/2012/11/Galaxy-for-NGS-Analysis-2012-10-09.pdf
 [29]: uploads/2013/07/Galaxy-for-NGS-Analysis-07-09-13.pdf
 [39]: uploads/2013/01/MS_variation_poster_final.pdf
 [30]: uploads/2013/01/Undergrad_poster.pdf
