<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
<div id="about"></div>

<div id="contact"></div>

<table style="width:100%;">
<tr>
    <td>
        <h1 style="text-align:left;">Matthew Shirley, Ph.D.</h1>
    </td>
    <td style="text-align:right;">
        <div class="hidden-print" align="right">
            <button type="button" class="btn btn-default" onclick="window.location.href = 'about?print=true'"><span class="glyphicon glyphicon-print"></span></button>
        </div>
    </td>
</tr>
</table>

<table style="width:100%;">
    <tr>
        <td style="text-align:left;vertical-align:top;padding-right:4em" width="60%">
            Rare disease and cancer genomics researcher, developer of software for genomics data.
        </td>

        <td style="text-align:right;vertical-align:top;" width="40%">
        250 Massachussetts Ave <br>
        Cambridge MA 02139 <br>
        <a href="mailto:mdshw5@gmail.com">mdshw5@gmail.com</a> <br>
        <i class="fa fa-mobile" style="color:#666666;font-size:80%"></i> (443) 637-2459 <br>
        <i class="fa fa-briefcase" style="color:#666666;font-size:80%"></i> (617) 871-5544 <br>
        <a href="http://mattshirley.com"><i class="far fa-compass"></i> mattshirley.com</a><br>
        <a href="https://impactstory.org/u/0000-0003-0855-9274"><i class="ai ai-impactstory"></i></a>
        <a href="https://twitter.com/mdshw5"><i class="fab fa-twitter-square"></i></a>
        <a href="https://github.com/mdshw5"><i class="fab fa-git-square"></i></a>
        <a href="https://linkedin.com/in/mdshw5"><i class="fab fa-linkedin"></i></a>
        <a href="http://depsy.org/person/352959"><i class="ai ai-depsy"></i></a> <br>
        </td>
    </tr>
</table>

<div id="education"></div>

#### Education
<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Ph.D. Biochemistry, Cellular, and Molecular Biology, 
        Johns Hopkins School of Medicine, Baltimore, MD</td></tr>
    <tr><td style="padding:2mm;"><strong>2008</strong></td><td>B.S. Biochemistry, University of Missouri, Columbia, MO</td></tr>
</table>

#### Research & Professional Experience

<table>
    <tr><td style="padding:2mm;"><strong>May 2015 - current</strong></td><td>Senior Expert II, Novartis Institutes for BioMedical Research, Cambridge MA</td></tr>
</table>

 - Developed and support an [alignment-free RNA-seq pipeline](https://github.com/Novartis/pisces) for processing thousands of internal and external data sets. 
 - Identified and validated novel pharmacodynamic markers to support multiple oncology drug development programs. 
 - Interface between wet lab biologists and sequencing facilities to guide experimental design, execution and perform data analysis of multiple NGS assays. 
 - Designed [new tools](#publications) for validation of CRISPR/Cas9 sgRNA efficiency. 
 - Mining internal and public data sets to identify or validate new drug targets. 
 - Organized a global computational research symposium spanning six geographical sites. 
 - Committee member of a global resource group for Novartis computational biology, bioinformatics and cheminformatics employees. 

<table>
    <tr><td style="padding:2mm;"><strong>Mar 2013 - May 2015</strong></td><td>Postdoctoral Fellow, Department of Oncology,
    Johns Hopkins School of Medicine</td></tr>
</table>

 - Developing statistical framework for integrative whole-genome genetic and bisulfite sequence analysis.
 - Mining genomic data for epigenetic prostate cancer biomarkers.
 - Collaboratively developing and deploying [new software tools](#software) in the genomics community.
 - Mentoring graduate and undergraduate students in computational methods.
 - Teaching [classes and workshops](#teaching) for computational biology and programming.

<table>
    <tr><td style="padding:2mm;"><strong>Aug 2008 - Mar 2013</strong></td><td>Ph.D. Candidate, Department of Neuroscience, Johns Hopkins School of Medicine</td></tr>
</table>

 - Discovered causative Sturge-Weber syndrome [somatic mutation](#publications) using whole genome sequencing.
 - Characterized [SNP and CNV variability in lymphoblast](#publications) cell lines using DNA microarrays.
 - Supported development of [software and method](#publications) for population-scale genetic relatedness testing.
 - Explored the human kinome for [novel kinase-substrate interactions](#publications) using protein microarrays.
 - Developed and deployed [web-based tool](#software) for somatic mosaicism detection.

<table>
    <tr><td style="padding:2mm;"><strong>Jun 2006 - May 2008</strong></td><td> Research Assistant, Department of Biochemistry,
 University of Missouri-Columbia</td></tr>
</table>

 - Studied genetics of heavy metal bioremediation in bacterium *Desulfovibrio vulgaris* Hildenborough.
 - Sequenced transposon insertion mutagenesis library in *DvH*.
 - Characterized metabolites of *DvH* anaerobic growth using gas chromatography.

#### Skills

<ul class="list-inline">
    <li>Sequencing Analysis</li>
    <li>Molecular Biology</li>
    <li>Applied Statistics</li>
    <li>Python</li>
    <li>R</li>
    <li>Bash</li>
    <li>Code Version Control</li>
    <li>SQL</li>
    <li>Continuous Integration Testing</li>
    <li>Unix</li>
    <li>Cloud Computing</li>
    <li>Software Development</li>
</ul>

<div id="publications" style="page-break-before:always;"></div>

#### Publications

<table>
    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1158/0008-5472.CAN-20-0189" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Hui Qin Wang, Iain J Mulford, Fiona Sharp, Jinsheng Liang, Sema Kurtulus, Gina Trabucco, David S Quinn, Tyler A Longmire, Nidhi Patel, Roshani Patil, <strong>Matthew D. Shirley</strong>, Yan Chen, Hao Wang, David A. Ruddy, Claire Fabre, Juliet A. Williams, Peter S. Hammerman, Jennifer Mataraza, Barbara Platzer and Ensar Halilovic. <a href="https://doi.org/10.1158/0008-5472.CAN-20-0189"><em>Inhibition of MDM2 promotes anti-tumor responses in p53 wild-type cancer cells through their interaction with the immune and stromal microenvironment</em></a>. Cancer Research. 2021.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1101/2020.12.01.390575" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Matthew D Shirley, Viveksagar K Radhakrishna, Javad Golji, Joshua M Korn. <a href="https://doi.org/10.1101/2020.12.01.390575"><em>(Preprint) PISCES: a package for rapid quantitation and quality control of large scale mRNA-seq datasets</em></a>. bioRxiv. 2020.</td></tr>
    
    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1158/1535-7163.MCT-19-1013 " data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Florencia Rago, GiNell Elliott, Ailing Li, Kathleen Sprouffske, Grainne Kerr, Aurore Desplat, Dorothee Abramowski, Julie T. Chen, Ali Farsidjani, Kay X. Xiang, Geoffrey Bushold, Yun Feng, <strong>Matthew D. Shirley</strong>, Anka Bric, Anthony Vattay, Henrik Möbitz, Katsumasa Nakajima, Christopher D. Adair, Simon Mathieu, Rukundo Ntaganda, Troy Smith, Julien P.N. Papillon, Audrey Kauffmann, David A. Ruddy, Hyo-eun C. Bhang, Deborah Castelletti and Zainab Jagani. <a href="10.1158/1535-7163.MCT-19-1013 "><em>The Discovery of SWI/SNF Chromatin Remodeling Activity as a Novel and Targetable Dependency in Uveal Melanoma</em></a>. Molecular Cancer Therapeutics. 2020.</td></tr>
    
      <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1158/1535-7163.MCT-19-0170" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Huai-Xiang Hao, Hongyun Wang, Chen Liu, Steven Kovats, Roberto Velazquez, Hengyu Lu, Bhavesh Pant, <strong>Matthew Shirley</strong>, Matthew J Meyer, Minying Pu, Joanne Lim, Michael Fleming, LeighAnn Alexander, Ali Farsidjani, Matthew J. LaMarche, Susan Moody, Serena J Silver, Giordano Caponigro, Darrin D. Stuart, Tinya J Abrams, Peter S. Hammerman, Juliet Williams, Jeffrey A. Engelman, Silvia Goldoni and Morvarid Mohseni. <a href="https://doi.org/10.1158/1535-7163.MCT-19-0170"><em>Tumor intrinsic efficacy by SHP2 and RTK inhibitors in KRAS mutant cancers</em></a>. Molecular Cancer Therapeutics. 2019.</td></tr>
      
      <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="https://doi.org/10.1101/812628" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Zainab Jagani, Gregg Chenail, Kay Xiang, Geoffrey Bushold, Hyo-Eun C. Bhang, Ailing Li, GiNell Elliott, Jiang Zhu, Anthony Vattay, Tamara Gilbert, Anka Bric, Rie Kikkawa, Valérie Dubost, Rémi Terranova, John Cantwell, Catherine Luu, Serena Silver, <strong>Matt Shirley</strong>, Francois Huet, Rob Maher, John Reece-Hoyes, David Ruddy, Daniel Rakiec, Joshua Korn, Carsten Russ, Vera Ruda, Julia Dooley, Emily Costa, Isabel Park, Henrik Moebitz, Katsumasa Nakajima, Christopher D. Adair, Simon Mathieu, Rukundo Ntaganda, Troy Smith, David Farley, Daniel King, Xiaoling Xie, Raviraj Kulathila, Tiancen Hu, Xuewen Pan, Qicheng Ma, Katarina Vulic, Florencia Rago, Scott Clarkson, Robin Ge, Frederic Sigoillot, Gwynn Pardee, Linda Bagdasarian, Margaret McLaughlin, Kristy Haas, Jan Weiler, Steve Kovats, Mariela Jaskelioff, Marie Apolline-Gerard, Johanna Beil, Ulrike Naumann, Pascal Fortin, Frank P. Stegmeier, Michael G. Acker, Juliet Williams, Matthew Meyer, James E. Bradner, Nicholas Keen, William R. Sellers, Francesco Hofmann, Jeffrey A. Engelman, Darrin Stuart, Julien P.N. Papillon. <a href="https://doi.org/10.1101/812628"><em>(Preprint) In-Depth Characterization and Validation in BRG1-Mutant Lung Cancers Define Novel Catalytic Inhibitors of SWI/SNF Chromatin Remodeling</em></a>. BioRxiv. 2019.</td></tr>      
    
    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1158/1541-7786.MCR-18-0327" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Grace P Leung, Tianshu Feng, Frederic D Sigoillot, Felipe Correa Geyer, <strong>Matthew D Shirley</strong>, David A Ruddy, Daniel Rakiec, Alyson K Freeman, Jeffrey A Engelman, Mariela Jaskelioff, Darrin D Stuart. <a href="https://doi.org/10.1158/1541-7786.MCR-18-0327"><em>Hyperactivation of MAPK Signaling is Deleterious to RAS/RAF Mutant Melanoma</em></a>. Molecular Cancer Research. 2018.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1158/2159-8290.CD-16-0178" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Munoz DM, Cassiani PJ, Li L, Billy E, Korn JM, Jones MD, Golji J, Ruddy DA, Yu K, McAllister G, DeWeck A, Abramowski D, Wan J, <strong>Shirley MD</strong>, Neshat SY, Rakiec D, de Beaumont R, Weber O, Kauffmann A, McDonald ER, Keen N, Hofmann F, Sellers WR, Schmelzle T, Stegmeier F, Schlabach MR. <a href="https://doi.org/10.1158/2159-8290.CD-16-0178"><em>CRISPR screens provide a comprehensive assessment of cancer vulnerabilities but generate false-positive hits for highly amplified genomic regions</em></a>. Cancer Discovery. 2016.</td></tr>
    
    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1371/journal.pone.0149646" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent"><strong>Matthew D Shirley</strong>, Laurence Frelin , José Soria López , Anne Jedlicka , Amanda Dziedzic , Michelle A. Frank-Crawford , Wayne Silverman , Louis Hagopian , Jonathan Pevsner. <a href="https://doi.org/10.1371/journal.pone.0149646"><em>Copy Number Variants Associated with 14 Cases of Self-Injurious Behavior</em></a>. PLOS ONE. 2016.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.7287/peerj.preprints.970v1" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent"><strong>Shirley MD</strong>, Ma Z, Pederson B, Wheelan S. <a href="https://doi.org/10.7287/peerj.preprints.970v1"><em>Efficient "pythonic" access to FASTA files using pyfaidx</em></a>. PeerJ PrePrints. 2015.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1056/NEJMoa1213507" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent"><strong>Shirley MD</strong>, Tang H, Gallione CJ, Baugher JD, Frelin LP, Cohen B, North PE, Marchuk DA, Comi AM, Pevsner J. <a href="https://doi.org/10.1056/NEJMoa1213507"><em>Sturge–Weber Syndrome and Port-Wine Stains Caused by Somatic Mutation in GNAQ</em></a>. New England Journal of Medicine. 2013.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1186/1471-2164-14-367" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Baugher JD, Baugher BD, <strong>Shirley MD</strong>, Pevsner J. <a href="https://doi.org/10.1186/1471-2164-14-367"><em>Sensitive and specific detection of mosaic chromosomal abnormalities using the Parent-of-Origin-based Detection (POD) method</em></a>. BMC Genomics. 2013.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1038/msb.2013.12" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">R.H. Newman, J. Hu, H. Rho, Z. Xie, C. Woodard, J. Neiswinger, C. Cooper, <strong>M. Shirley</strong> et al.
        <a href="https://doi.org/10.1038/msb.2013.12"><em>Construction of human activity-based phosphorylation networks</em></a>. Molecular Systems Biology. 2013.</td></tr>

        <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1371/journal.pone.0049575" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent">Stevens EL, Baugher JD, <strong>Shirley MD</strong>, Frelin LP, Pevsner J. 
    <a href="https://doi.org/10.1371/journal.pone.0049575"><em>Unexpected Relationships and Inbreeding in HapMap Phase III Populations</em></a>. PLOS ONE. 2012.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1002/humu.22062" data-hide-no-mentions="true" class="hidden-print altmetric-embed"></div></td><td class="hangingindent"><strong>Shirley MD</strong>, Baugher JD, Stevens EL, Tang Z, Gerry N, Beiswanger CM, Berlin DS, Pevsner J. 
        <a href="uploads/2012/10/Chromosomal-Variation-in-Lymphoblastoid-Cell-Lines.pdf"><em>Chromosomal variation in lymphoblastoid cell lines</em></a>. Human Mutation. 2012.</td></tr>
</table>  
<div id="software"></div>

#### Software

<table>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=Novartis&repo=pisces&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/Novartis/pisces">PISCES</a>: a package for rapid quantitation and quality control of large scale mRNA-seq datasets</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=pyfaidx&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/pyfaidx">pyfaidx</a>: efficient & pythonic random access to fasta subsequences</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=simplesam&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/simplesam">simplesam</a>: Simple pure Python SAM parser and objects for working with SAM records</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=fastqp&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/fastqp">fastqp</a>: Python FASTQ and SAM read quality assessment and plotting</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=sra-tools-galaxy&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/sra-tools-galaxy">sra-tools-galaxy</a>: NCBI SRA toolkit Galaxy package and tool wrappers</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=hamstring&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/hamstring">hamstring</a>: Python module for Hamming7,4 error-correcting DNA barcodes</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=jdbaugher&repo=tripod&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="http://tripod.mattshirley.com">tripod</a>: web interface for the triPOD mosaicism analysis</td></tr>
</table>

#### Patents

<table>
    <tr><td style="vertical-align:top"><td class="hangingindent">Jonathan Pevsner, Anne Comi, Douglas Marchuk, <strong>Matthew Shirley</strong>
        <a href="http://patft1.uspto.gov/netacgi/nph-Parser?patentnumber=10544461"><em>Diagnostic and prognostic test for Sturge-Weber syndrome, Klippel-Trenaunay-Weber syndrome, and port-wine stains (PWSS)</em></a>. USPTO. 2020.</td></tr>
</table>

<div id="grants"></div>

#### Grants

<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Amazon Web Services in Education Grant, supporting <em>Introduction to Python</em></td></tr>
</table>

<div id="teaching"></div>

#### Teaching Experience

<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD</td></tr>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Bioinformatics & Computational Biology Department, Foundation for Advanced Education in the Sciences, Bethesda MD</td></tr>
</table>
