<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
<div id="about"></div>

<div id="contact"></div>

<table style="width:100%;">
<tr>
    <td>
        <h1 style="text-align:left;">Matthew Shirley, Ph.D.</h1>
    </td>
    <td style="text-align:right;">
        <div class="hidden-print" align="right">
            <button type="button" class="btn btn-default" onclick="window.location.href = 'about?print=true'"><span class="glyphicon glyphicon-print"></span></button>
        </div>
    </td>
</tr>
</table>

<table style="width:100%;">
    <tr>
        <td style="text-align:left;vertical-align:top;padding-right:4em" width="60%">
            Rare disease and prostate cancer genomics researcher, developer of software for genomics data.
        </td>

        <td style="text-align:right;vertical-align:top;" width="40%">
        Sidney Kimmel Comprehensive Cancer Center <br>
        CRBII 142 1550 Orleans Street <br>
        Baltimore MD 21287 <br>
        <a href="mailto:mdshw5@gmail.com">mdshw5@gmail.com</a> <br>
        <i class="fa fa-briefcase" style="color:#666666;font-size:80%"></i> (443) 637-2459 <br>
        <a href="https://impactstory.org/MatthewShirley"><img src="https://impactstory.org/static/img/impactstory-logo.png" width="15" /></a>
        <a href="https://twitter.com/mdshw5"><i class="fa fa-twitter"></i></a>
        <a href="https://github.com/mdshw5"><i class="fa fa-git"></i></a>
        <a href="https://linkedin.com/in/mdshw5"><i class="fa fa-linkedin-square"></i></a> <br>
        </td>
    </tr>
</table>

<div id="education"></div>

#### Education
<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Ph.D. Biochemistry, Cellular, and Molecular Biology, 
        Johns Hopkins School of Medicine, Baltimore, MD</td></tr>
    <tr><td style="padding:2mm;"><strong>2008</strong></td><td>B.S. Biochemistry, University of Missouri, Columbia, MO</td></tr>
</table>

#### Research Experience

<table>
    <tr><td style="padding:2mm;"><strong>2013-</strong></td><td>Postdoctoral Fellow, Department of Oncology,
    Johns Hopkins School of Medicine</td></tr>
</table>

 - Developing computational tools for novel bisulfite sequencing methods.
 - Mining NGS data for epigenetic prostate cancer biomarkers.
 - Creating visualizations for exploring geometric relationships between genomic elements.
 - Collaboratively developing and deploying [new tools](#software) in the bioinformatics community.
 - Teaching [classes and workshops](#teaching) for computational biology and programming.

    *Mentors: Dr. Sarah Wheelan and Dr. Vasan Yegnasubramanian*  

<table>
    <tr><td style="padding:2mm;"><strong>2008-2013</strong></td><td>Ph.D. Candidate, Department of Neuroscience, Johns Hopkins School of Medicine</td></tr>
</table>

 - Discovered [somatic mutation](#publications) causing Sturge-Weber syndrome using whole genome sequencing.
 - Characterized [SNP and CNV variability in lymphoblast](#publications) cell lines using DNA microarrays.
 - Supported development of [software and method](#publications) for population-scale genetic relatedness testing.
 - Explored the human kinome for [novel kinase-substrate interactions](#publications) using protein microarrays.
 - Developed and deployed [web-based tool](#software) for somatic mosaicism detection.

    *Mentors: Dr. Jonathan Pevsner, Dr. Jin Zhang, and Dr. Heng Zhu*

<table>
    <tr><td style="padding:2mm;"><strong>2006-2008</strong></td><td> Research Assistant, Department of Biochemistry,
 University of Missouri-Columbia</td></tr>
</table>

 - Studied genetics of heavy metal bioremediation in bacterium *Desulfovibrio vulgaris* Hildenborough.
 - Sequenced transposon insertion mutagenesis library in *DvH*.
 - Characterized metabolites of *DvH* anaerobic growth using gas chromatography.

    *Mentor: Dr. Judy Wall*

#### Skills

<ul class="list-inline">
    <li>Sequencing Analysis</li>
    <li>Molecular Biology</li>
    <li>Data Exploration</li>
    <li>Python</li>
    <li>R</li>
    <li>Version Control</li>
    <li>Continuous Integration</li>
    <li>Unix</li>
    <li>Cluster/Cloud Computing</li>
    <li>Software Development</li>
    <li>Web UI Design</li>
    <li>Javascript</li>
    <li>SQL</li>
</ul>

<div id="publications" style="page-break-before:always;"></div>

#### Peer Reviewed Publications

<table>
    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1056/NEJMoa1213507" data-hide-no-mentions="true" class="altmetric-embed"></div></td><td><strong>Shirley MD</strong>, Tang H, Gallione CJ, Baugher JD, Frelin LP, Cohen B, North PE, Marchuk DA, Comi AM, Pevsner J. <a href="uploads/2012/03/N-Engl-J-Med-2013-Shirley.pdf"><em>Sturge–Weber Syndrome and Port-Wine Stains Caused by Somatic Mutation in GNAQ</em></a>. New England Journal of Medicine. 2013.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1038/msb.2013.12" data-hide-no-mentions="true" class="altmetric-embed"></div></td><td>R.H. Newman, J. Hu, H. Rho, Z. Xie, C. Woodard, J. Neiswinger, C. Cooper, <strong>M. Shirley</strong> et al.
        <a href="uploads/2013/04/Mol-Syst-Biol-2013-Newman.pdf"><em>Construction of human activity-based phosphorylation networks</em></a>. Molecular Systems Biology. 2013.</td></tr>

        <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1371/journal.pone.0049575" data-hide-no-mentions="true" class="altmetric-embed"></div></td><td>Stevens EL, Baugher JD, **Shirley MD**, Frelin LP, Pevsner J. 
    <a href="uploads/2013/03/journal.pone_.0049575.pdf"><em>Unexpected Relationships and Inbreeding in HapMap Phase III Populations</em></a>. PLoS ONE. 2012.</td></tr>

    <tr><td style="vertical-align:top"><div data-badge-popover="top" data-badge-type="donut" data-doi="10.1002/humu.22062" data-hide-no-mentions="true" class="altmetric-embed"></div></td><td><strong>Shirley MD</strong>, Baugher JD, Stevens EL, Tang Z, Gerry N, Beiswanger CM, Berlin DS, Pevsner J. 
        <a href="uploads/2012/10/Chromosomal-Variation-in-Lymphoblastoid-Cell-Lines.pdf"><em>Chromosomal variation in lymphoblastoid cell lines</em></a>. Human Mutation. 2012.</td></tr>
</table>  
<div id="software"></div>

#### Software

<table>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=pyfaidx&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/pyfaidx">pyfaidx</a>: efficient & pythonic random access to fasta subsequences.</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=fastqp&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/fastqp">fastqp</a>: simple Python FASTQ and SAM read quality assessment and plotting.</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=mdshw5&repo=hamstring&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="https://github.com/mdshw5/hamstring">hamstring</a>: Python module for generating / demultiplexing / correcting Hamming7,4 error-correcting DNA barcodes.</td></tr>
    <tr><td style="vertical-align:top"><iframe src="http://ghbtns.com/github-btn.html?user=jdbaugher&repo=tripod&type=watch&count=true" allowtransparency="true" frameborder="0" scrolling="0" width="90" height="20"></iframe></td><td><a href="http://tripod.mattshirley.com">tripod</a>: web interface for the triPOD mosaicism analysis.</td></tr>
    <tr></td><td style="vertical-align:top"><td><a href="http://toolshed.g2.bx.psu.edu/view/matt-shirley/ncbi_sra_toolkit">sra toolkit</a>: Galaxy tool definitions for <a href="http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software">NCBI SRA Toolkit</a></td></tr>
</table>

<div id="grants"></div>

#### Grants

<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Amazon Web Services in Education Grant, supporting <em>Introduction to Python</em></td></tr>
</table>

<div id="teaching"></div>

#### Teaching Experience

<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD</td></tr>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Bioinformatics & Computational Biology Department, Foundation for Advanced Education in the Sciences, Bethesda MD</td></tr>
</table>
