<div id="about"></div>

<div id="contact"></div>

<table style="width:100%;">
<tr>
    <td>
        <h1 style="text-align:left;">Matthew Shirley, Ph.D.</h1>
    </td>
    <td style="text-align:right;">
        <div class="hidden-print" align="right">
            <button type="button" class="btn btn-default" onclick="window.location.href = 'about?print=true&resume=seres_health_resume'"><span class="glyphicon glyphicon-print"></span></button>
        </div>
    </td>
</tr>
</table>

<table style="width:100%;">
    <tr>
        <td style="text-align:left;vertical-align:top;">
        <p>Computational Biologist<br>
        Developing bioinformatics software for high-dimensional genomics data.
        </td>

        <td style="text-align:right;vertical-align:top;">
        1900 Sulgrave Ave Apt 2E <br>
        Baltimore, MD 21209 <br>
        <a href="mailto:mdshw5@gmail.com">mdshw5@gmail.com</a> <br>
        <i class="fa fa-briefcase" style="color:#666666;font-size:80%"></i> (443) 637-2459 <br>
        <a href="https://twitter.com/mdshw5"><i class="fa fa-twitter"></i></a>
        <a href="https://github.com/mdshw5"><i class="fa fa-git"></i></a>
        <a href="https://linkedin.com/in/mdshw5"><i class="fa fa-linkedin-square"></i></a> <br>
        </td>
    </tr>
</table>

<div id="education"></div>

#### Education
<table>
    <tr><td style="padding:2mm;"><strong>2013</strong></td><td>Ph.D. Biochemistry, Cellular, and Molecular Biology, 
        Johns Hopkins School of Medicine, Baltimore MD</td></tr>
    <tr><td style="padding:2mm;"><strong>2008</strong></td><td>B.S. Biochemistry, University of Missouri, Columbia MO</td></tr>
</table>

#### Research Experience

**2013-current**: Postdoctoral Fellow, Department of Oncology,
    Johns Hopkins School of Medicine

 - Developing computational tools for novel bisulfite sequencing methods.
 - Mining NGS data for epigenetic prostate cancer biomarkers.
 - Collaboratively developing and deploying [new tools](#software) in the bioinformatics community.
 - Teaching classes and workshops for computational biology and programming.

    *Mentors: Dr. Sarah Wheelan and Dr. Vasan Yegnasubramanian*  

**2008-2013**: Ph.D. Candidate, Department of Neuroscience, Johns Hopkins School of Medicine

 - Discovered [somatic mutation](#publications) causing Sturge-Weber syndrome using whole genome sequencing.
 - Characterized [SNP and CNV variability in lymphoblast](#publications) cell lines using DNA microarrays.
 - Supported development of [software and method](#publications) for population-scale genetic relatedness testing.
 - Interrogated the kinome for [novel kinase-substrate interactions](#publications) using protein microarrays.
 - Developed and deployed [web-based tool](#software) for somatic mosaicism detection.

    *Mentors: Dr. Jonathan Pevsner, Dr. Jin Zhang, and Dr. Heng Zhu*


**2006-2008**: Research Assistant, Department of Biochemistry,
 University of Missouri-Columbia

 - Studied heavy metal bioremediation potential of bacterium *Desulfovibrio vulgaris* Hildenborough.
 - Sequenced transposon insertion mutagenesis library in *DvH*.
 - Characterized metabolites of *DvH* anaerobic growth using gas chromatography.

    Mentor: *Dr. Judy Wall*

#### Skills

<ul class="list-inline">
    <li>NGS Experimental Design</li>
    <li>DNA/RNA/ChIP-seq Analysis</li>
    <li>Statistical Data Exploration</li>
    <li>Machine Learning</li>
    <li>Data Management</li>
    <li>R</li>
    <li>Python</li>
    <li>Bash</li>
    <li>Unix</li>
    <li>Software Development</li>
    <li>Full-stack Web Development</li>
    <li>Cluster/Cloud Computing</li>
    <li></li>
</ul>

<div id="publications" style="page-break-before:always;"></div>

#### Peer Reviewed Publications

- **Shirley MD**, Tang H, Gallione CJ, Baugher JD, Frelin LP, Cohen B, North PE, Marchuk DA, Comi AM, Pevsner J.
  *[Sturge–Weber Syndrome and Port-Wine Stains Caused by Somatic Mutation in GNAQ][3]*.
  New England Journal of Medicine. 2013. PMID: [23656586][4].  

- R.H. Newman, J. Hu, H. Rho, Z. Xie, C. Woodard, J. Neiswinger, C. Cooper, **M. Shirley** et al.
  *[Construction of human activity-based phosphorylation networks][13]*.
  Molecular Systems Biology. 2013. PMID: [23549483][14].

- Stevens EL, Baugher JD, **Shirley MD**, Frelin LP, Pevsner J. 
  *[Unexpected Relationships and Inbreeding in HapMap Phase III Populations][16]*.
  PLoS ONE. 2012. PMID: [23185369][17].

- **Shirley MD**, Baugher JD, Stevens EL, Tang Z, Gerry N, Beiswanger CM, Berlin DS, Pevsner J. 
  *[Chromosomal variation in lymphoblastoid cell lines][18]*. 
  Human Mutation. 2012. PMID: [22374857][19].

<div id="software"></div>

#### Software

- [pyfaidx](https://github.com/mdshw5/pyfaidx): efficient & pythonic random access to fasta subsequences.
- [fastqp](https://github.com/mdshw5/fastqp): simple FASTQ and SAM read quality assessment and plotting using Python.
- [hamstring][9]: Python module for generating Hamming7,4 error-correcting DNA barcodes.
- [triPOD][11]: web interface for the triPOD mosaicism analysis.
- [sra toolkit](http://toolshed.g2.bx.psu.edu/view/matt-shirley/ncbi_sra_toolkit): Galaxy tool definitions for [NCBI SRA Toolkit](http://www.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software).
- [ncbi epigenomics](http://toolshed.g2.bx.psu.edu/view/matt-shirley/ncbi_epi_browse): Galaxy interface to [NCBI Epigenomics](http://www.ncbi.nlm.nih.gov/epigenomics).

<div id="grants"></div>

#### Grants
 | |
---- | - |-------------------
**2013** |  | Amazon Web Services in Education Grant, supporting *Introduction to Python*

<div id="teaching"></div>

#### Teaching Experience
| |
---- | - | -------------------
**2013** | | Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD
**2013** | | Bioinformatics & Computational Biology Department, [FAES](http://faes.org), Bethesda MD

 [1]: http://biolchem.bs.jhmi.edu/bcmb/Pages/index.aspx
 [2]: http://biochem.missouri.edu/
 [3]: http://mattshirley.com/uploads/2012/03/N-Engl-J-Med-2013-Shirley.pdf
 [4]: http://www.ncbi.nlm.nih.gov/pubmed/23656586
 [5]: http://www.sturge-weber.org/component/content/article/4-general-info/311-route-to-a-cure.html
 [6]: http://www.sturge-weber.org/images/stories/BVMC/explaining_gnaq.pdf
 [7]: http://www.sturge-weber.org/images/stories/BVMC/sturge-weber%20foundation%20nejm%20news%20release%20final%20may-08-2013.pdf
 [8]: http://faes.org
 [9]: http://mdshw5.github.io/hamstring/
 [10]: http://www.plosone.org/article/info:doi/10.1371/journal.pone.0036852
 [11]: http://tripod.mattshirley.com
 [12]: http://www.biomedcentral.com/1471-2164/14/367/
 [13]: http://mattshirley.com/uploads/2013/04/Mol-Syst-Biol-2013-Newman.pdf
 [14]: http://www.ncbi.nlm.nih.gov/pubmed/?term=23549483
 [15]: http://www.nature.com/doifinder/10.1038/msb.2013.12
 [16]: http://mattshirley.com/uploads/2013/03/journal.pone_.0049575.pdf
 [17]: http://www.ncbi.nlm.nih.gov/pubmed/23185369
 [18]: http://mattshirley.com/uploads/2012/03/22062_ftp.pdf
 [19]: http://www.ncbi.nlm.nih.gov/pubmed/22374857
 [21]: http://mattshirley.com/uploads/2013/05/SWS_thesis_seminar.html
 [22]: http://mattshirley.com/uploads/2013/04/lecture14-notes.html
 [23]: http://www.faes.org
 [24]: http://mattshirley.com/uploads/2013/04/lecture13-notes.html
 [25]: http://mattshirley.com/uploads/2013/04/lecture12-notes.html
 [26]: http://mattshirley.com/uploads/2013/02/Galaxy-for-NGS-Analysis.pdf
 [27]: http://mattshirley.com/uploads/2012/10/Chromosomal-Variation-in-Lymphoblastoid-Cell-Lines.pdf
 [28]: http://mattshirley.com/uploads/2012/11/Galaxy-for-NGS-Analysis-2012-10-09.pdf
 [29]: http://mattshirley.com/uploads/2013/07/Galaxy-for-NGS-Analysis-07-09-13.pdf
 [39]: http://mattshirley.com/uploads/2013/01/MS_variation_poster_final.pdf
 [30]: http://mattshirley.com/uploads/2013/01/Undergrad_poster.pdf
