#### Lectures

<h5>Titles of talks link to slides.</h5>

- BIOF309: Introduction to Python [Working with Pandas](http://mattshirley.com/uploads/2014/05/working_with_pandas.slides.html), May 1, 2014, FAES
- BIOF309: Introduction to Python [Sequence alignment with Biopython](http://mattshirley.com/uploads/2014/04/sequence_alignment_biopython.slides.html), April 24, 2014, FAES
- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][29], April 8, 2014, FAES
- Introduction to Python: [data structures and flow control](uploads/2013/10/Introduction%20to%20Python%20Day%201.ipynb.slides.html), October 28 2013, CCG
- Introduction to Python: [functions, classes, and I/O](uploads/2013/10/Introduction%20to%20Python%20Day%202.ipynb.slides.html), October 30 2013, CCG
- BIOF503: Make Striking Figures and Analyze Data with R, [Spoon-Fed R](uploads/2013/10/Spoon-fed_R.html), October 24 2013, FAES
- BIOF309: Introduction to Python, [Interfacing R and Python using Rpy2][22], May 2 2013, FAES
- BIOF309: Introduction to Python, [Hello web: Accessing databases and APIs using Biopython][24], April 25 2013, FAES
- BIOF309: Introduction to Python, [Reading, processing, and writing high throughput sequencing data][25], April 18 2013, FAES
- BIOF521: Bioinformatics for analysis of data generated by NGS, [Galaxy for Next Generation Sequencing Analysis][26], February 6 2013, FAES
- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][29], July 9 2013, FAES
- Biotrac45: Bioinformatic Analysis of Next Generation Sequencing Data, [Galaxy for Next Generation Sequencing Analysis][28], November 12 2012, FAES

<div id="seminars"></div>

#### Seminars
- [*PISCES: alignment free RNA-seq quantiation and QC pipeline*](http://www.openwetware.org/images/6/6b/2017-02-16_BIG-meeting.pdf), February 16 2017, [MIT BIG meeting](http://www.openwetware.org/wiki/BioMicroCenter:BIG_meeting), MIT, Cambridge MA ([video](https://www.youtube.com/watch?v=6uS5JENyDg0))
- [*Deep amplicon sequencing reveals GNAQ 548G>A as the causal somatic mutation in Sturge-Weber syndrome and common port-wine stains*](uploads/2013/10/Young-Investigators-Seminar.pdf), October 17 2013, Center for Computational Genomics, Johns Hopkins School of Medicine, Baltimore MD
- [*Chromosomal Variation in Lymphoblastoid Cell Lines*][27], December 10 2012, Department of Biochemistry, Cellular, and Molecular Biology, Johns Hopkins School of Medicine, Baltimore MD

#### Abstracts and Posters

- **Shirley MD** et al. *[A Somatic Mosaic Mutation in GNAQ causes Sturge-Weber syndrome and Isolated Port-wine Birthmarks](http://mattshirley.com/uploads/2014/04/ana24069.pdf)*, 42nd Annual Meeting of the Child Neurology Society. November 2013
- **Shirley MD** et al. [Chromosomal Variation in Lymphoblastoid Cell Lines][39], April 25 2012, Johns Hopkins School of Medicine, Baltimore MD
- **Shirley MD** et al. [Rapid Automated Characterization of Transposon Insertion Mutants in *Desulfovibrio vulgaris* Hildenborough by srnPCR][30], Undergraduate Research and Creative Achievements Forum, 2007, University of Missouri, Columbia MO
- **Shirley MD** et al. [Rapid Automated Characterization of Transposon Insertion Mutants in *Desulfovibrio vulgaris* Hildenborough by srnPCR](http://vimss.lbl.gov/publicfiles/ESPP_retreat_2007_Shirley.pdf), Virtual Institute for Microbial Stress and Survival, 2007, Lawrence Berkeley National Laboratories, Berkeley CA

<div id="awards"></div>

#### Awards
| |
---- | - |-------------------
**2007** | | Undergraduate Research Ambassador
**2007** | | Chancellor’s Award for Excellence in Research


 [1]: http://biolchem.bs.jhmi.edu/bcmb/Pages/index.aspx
 [2]: http://biochem.missouri.edu/
 [3]: http://mattshirley.com/uploads/2012/03/N-Engl-J-Med-2013-Shirley.pdf
 [4]: http://www.ncbi.nlm.nih.gov/pubmed/23656586
 [5]: http://www.sturge-weber.org/component/content/article/4-general-info/311-route-to-a-cure.html
 [6]: http://www.sturge-weber.org/images/stories/BVMC/explaining_gnaq.pdf
 [7]: http://www.sturge-weber.org/images/stories/BVMC/sturge-weber%20foundation%20nejm%20news%20release%20final%20may-08-2013.pdf
 [8]: http://faes.org
 [9]: http://mdshw5.github.io/hamstring/
 [10]: http://www.plosone.org/article/info:doi/10.1371/journal.pone.0036852
 [11]: http://tripod.mattshirley.com
 [12]: http://www.biomedcentral.com/1471-2164/14/367/
 [13]: http://mattshirley.com/uploads/2013/04/Mol-Syst-Biol-2013-Newman.pdf
 [14]: http://www.ncbi.nlm.nih.gov/pubmed/?term=23549483
 [15]: http://www.nature.com/doifinder/10.1038/msb.2013.12
 [16]: http://mattshirley.com/uploads/2013/03/journal.pone_.0049575.pdf
 [17]: http://www.ncbi.nlm.nih.gov/pubmed/23185369
 [18]: http://mattshirley.com/uploads/2012/03/22062_ftp.pdf
 [19]: http://www.ncbi.nlm.nih.gov/pubmed/22374857
 [21]: http://mattshirley.com/uploads/2013/05/SWS_thesis_seminar.html
 [22]: http://mattshirley.com/uploads/2013/04/lecture14-notes.html
 [23]: http://www.faes.org
 [24]: http://mattshirley.com/uploads/2013/04/lecture13-notes.html
 [25]: http://mattshirley.com/uploads/2013/04/lecture12-notes.html
 [26]: http://mattshirley.com/uploads/2013/02/Galaxy-for-NGS-Analysis.pdf
 [27]: http://mattshirley.com/uploads/2012/10/Chromosomal-Variation-in-Lymphoblastoid-Cell-Lines.pdf
 [28]: http://mattshirley.com/uploads/2012/11/Galaxy-for-NGS-Analysis-2012-10-09.pdf
 [29]: http://mattshirley.com/uploads/2013/07/Galaxy-for-NGS-Analysis-07-09-13.pdf
 [39]: http://mattshirley.com/uploads/2013/01/MS_variation_poster_final.pdf
 [30]: http://mattshirley.com/uploads/2013/01/Undergrad_poster.pdf
 
